### PR TITLE
NEO-1526: `--small` Select variant

### PR DESCRIPTION
**Pull request by mrazian85**
[Link to updated components in Deploy Preview](https://deploy-preview-174--neo-react-library-storybook.netlify.app/?path=/story/components-select--small-selects)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

Tweaked styling for `--small` Select variant, added logic to Internal Select Components to ensure correct application of related classes and styles.

As per the note in [NEO-1526](https://jira.forge.avaya.com/browse/NEO-1526), I decided it would be best to resolve this issue in the `Select_shim.css` file inside the React library itself, as it involves class names that are only used there, ex. `.neo-multiselect__padded-container`

### CODE REVIEW

File: src/components/Select/InternalComponents/MultiSelectSearchable.tsx
Comment (joe-s-avaya) [inline src/components/Select/InternalComponents/MultiSelectSearchable.tsx:1]: you have to add `*--small*` _and_ styles? that doesn't seem right...

File: src/components/Select/Select_shim.css [hunk 1]
```diff
@@ -79,6 +82,55 @@ div.neo-multiselect button.neo-multiselect__header {
...
+  .neo-chip {
+  min-height: initial;
+  max-height: 18px;
+  margin: 0px 2px;
+}
+
+.neo-multiselect.neo-multiselect--small.neo-multiselect--small-searchable:focus-within
+  .neo-multiselect-combo__header,
+.neo-multiselect.neo-multiselect--small:focus-within .neo-multiselect__header {
+  margin: 0 -1px;
```
Comment (joe-s-avaya) [inline src/components/Select/Select_shim.css:126]: negative values are a yellow flag, and negative _one_ pixel? that smells...

**Review by enrique-prado — APPROVED** [review]

**Review by joe-s-avaya — APPROVED** [review]
much cleaner, nice!

**Comment by matthewAvaya** [comment]
I'm not able to navigate through the list of dropdown items using my keyboard for the first small select component named 'Select a favorite food'. 
Screen Shot 2023-03-31 at 12 42 24 PM

Also some accessibility issues on the screen reader, but let me know if this PR doesn't need to touch on this.

**Comment by mrazian85** [comment]
> I'm not able to navigate through the list of dropdown items using my keyboard for the first small select component named 'Select a favorite food'. Screen Shot 2023-03-31 at 12 42 24 PM
> 
> Also some accessibility issues on the screen reader, but let me know if this PR doesn't need to touch on this.

Hey @matthewAvaya ; yup, this PR makes no changes to the Select other than cosmetic/styling. I know there are a couple a11y-related tickets open for the Select already; I'll make a note to add the keyboard behaviour to an appropriate one if it exists, otherwise will make a new one.

**Review by matthewAvaya — APPROVED** [review]

**Pull request merged**